Src/ILGPU/Backends/OpenCL/CLCodeGenerator.Values.cs

```diff
@@ -569,6 +569,52 @@ public void GenerateCode(SetField value)
                 value,
                 CLInstructions.GetLaneIndexOperation);
 
+        /// <summary cref="IBackendCodeGenerator.GenerateCode(DynamicMemoryLengthValue)"/>
+        public void GenerateCode(DynamicMemoryLengthValue value)
+        {
+            if (value.AddressSpace != MemoryAddressSpace.Shared)
+                throw new InvalidCodeGenerationException();
+
+            // Resolve the name of the global variable containing the length of the
+            // shared dynamic memory buffer.
+            var dynamicView = value.GetFirstUseNode().ResolveAs<Alloca>();
```

Comment:
Question: Currently, the invariant seems to be the case that the only occurrence of the `DynamicMemoryLengthValue` is the construction of an allocation. Is this guaranteed in all cases? Is there the possibility that we will add another pass in the future that could "rewire" the uses of this node to some other operation?

Reply:
For now, it's guaranteed. If we use `DynamicMemoryLengthValue` somewhere else in the IR generation, then that could cause issues.

The problem is that `DynamicMemoryLengthValue` is used to construct the `Alloca`, so we cannot pass the ID of the IR node. If we could, then we could decouple the `CLCodeGenerator` from relying on `Alloca`.